astropy/wcs/wcsapi/fitswcs.py

```diff
@@ -330,12 +332,93 @@ def _get_components_and_classes(self):
                 components[self.wcs.lng] = ('celestial', 0, 'spherical.lon.degree')
                 components[self.wcs.lat] = ('celestial', 1, 'spherical.lat.degree')
 
-        # Fallback: for any remaining components that haven't been identified, just
-        # return Quantity as the class to use
+        # We can then make sure we correctly return Time objects where appropriate
+        # (https://www.aanda.org/articles/aa/pdf/2015/02/aa24653-14.pdf)
 
         if 'time' in self.world_axis_physical_types:
-            warnings.warn('In future, times will be represented by the Time class '
-                          'instead of Quantity', FutureWarning)
+
+            multiple_time = self.world_axis_physical_types.count('time') > 1
+
+            for i in range(self.naxis):
+
+                if self.world_axis_physical_types[i] == 'time':
+
+                    if multiple_time:
+                        name = f'time.{i}'
+                    else:
+                        name = 'time'
+
+                    # Initalize JD offsets
+                    mjd1 = 0.
+                    mjd2 = 0.
+
+                    # Extract time scale
+                    scale = self.wcs.ctype[i].lower()
+
+                    # TODO: consider having GPS as a scale in Time
+                    # For now GPS is not a scale, we approximate this by TAI - 19s
```

Comment:
You can do this by also including `format`, so use `scale='tai'` and `format='gps'` - though that actually doesn't allow you to put `format='mjd'` below, so never mind...

Reply:
Yeah I went through the same reasoning :) I'll mark this as resolved.